src/mmw/js/src/modeling/models.js

```diff
@@ -351,7 +351,7 @@ var ProjectModel = Backbone.Model.extend({
 
                     pollFailure: function(err) {
                         console.log('Failed to gather data required for MAPSHED');
-                        promise.reject(err);
+                        promise.reject(taskModel.get('error'));
```

Comment:
Consider reverting this change. The promise should faithfully reject with the original error, and any presentational labeling should happen at the very end. There may be other listeners depending on this promise's rejection, for example.